### ISSUE TITLE
Error when calling completion-at-point following a successful completion

### ISSUE THREAD

**Issue by mcwitt** [joaotavora/eglot#1454]
In some contexts, calling `completion-at-point` twice in a row, or trying to complete with the point immediately after a valid symbol, results in an error. Observed with multiple language server backends (Python/pyright, Nix/nil, Haskell/haskell-language-server).

- Server used: pyright
- Emacs version: GNU Emacs 30.0.91
- Operating system: Linux
- Eglot version: 1.17
- Eglot installation method: use-package
- Using Doom: no

### MWE

1. `emacs -Q -f package-initialize` (or just `emacs -Q`)
2. C-x C-f foo.py
3. M-x eglot
4. Type "import pathl"
5. M-TAB (completes to "pathlib")
6. M-TAB again
7. The error message `Wrong type argument: sequencep, t` is displayed in the minibuffer

### Backtrace
<details>

```
Debugger entered--Lisp error: (wrong-type-argument sequencep t)
  eglot--dumb-tryc(#("pathlib" 0 7 (fontified t)) #f(compiled-function (pattern pred action) #<bytecode 0x1e99b8b955158244>) nil 7)
  #f(compiled-function (style) #<bytecode -0xeb3a97c58b68a20>)(eglot--dumb-flex)
  #f(compiled-function (elt) #<bytecode 0x1149fb86e70c59ab>)(eglot--dumb-flex)
  mapc(#f(compiled-function (elt) #<bytecode 0x1149fb86e70c59ab>) (eglot--dumb-flex basic partial-completion emacs22))
  seq-do(#f(compiled-function (elt) #<bytecode 0x1149fb86e70c59ab>) (eglot--dumb-flex basic partial-completion emacs22))
  seq-some(#f(compiled-function (style) #<bytecode -0xeb3a97c58b68a20>) (eglot--dumb-flex basic partial-completion emacs22))
  completion--nth-completion(1 #("pathlib" 0 7 (fontified t)) #f(compiled-function (pattern pred action) #<bytecode 0x1e99b8b955158244>) nil 7 (metadata (category . eglot-capf) (display-sort-function . #<subr F616e6f6e796d6f75732d6c616d626461_anonymous_lambda_160>)))
  completion-try-completion(#("pathlib" 0 7 (fontified t)) #f(compiled-function (pattern pred action) #<bytecode 0x1e99b8b955158244>) nil 7 (metadata (category . eglot-capf) (display-sort-function . #<subr F616e6f6e796d6f75732d6c616d626461_anonymous_lambda_160>)))
  completion--do-completion(#<marker at 8 in foo.py> 15)
  completion--in-region-1(#<marker at 8 in foo.py> 15)
  #f(compiled-function (start end collection predicate) #<bytecode 0x109f75683a8dc506>)(#<marker at 8 in foo.py> 15 #f(compiled-function (pattern pred action) #<bytecode 0x1e99b8b955158244>) nil)
  apply(#f(compiled-function (start end collection predicate) #<bytecode 0x109f75683a8dc506>) (#<marker at 8 in foo.py> 15 #f(compiled-function (pattern pred action) #<bytecode 0x1e99b8b955158244>) nil))
  #f(compiled-function (funs global args) #<bytecode 0x1a98a244d6119d5c>)(nil nil (#<marker at 8 in foo.py> 15 #f(compiled-function (pattern pred action) #<bytecode 0x1e99b8b955158244>) nil))
  completion--in-region(#<marker at 8 in foo.py> 15 #f(compiled-function (pattern pred action) #<bytecode 0x1e99b8b955158244>) nil)
  completion-in-region(#<marker at 8 in foo.py> 15 #f(compiled-function (pattern pred action) #<bytecode 0x1e99b8b955158244>) nil)
  completion-at-point()
  funcall-interactively(completion-at-point)
  command-execute(completion-at-point)
```
</details>

### M-x eglot-events-buffer
<details>

```
[jsonrpc] D[17:23:53.221] Running language server: /nix/store/3n04wicy6jpks8qbsqdkjfck1zdwqfpa-pyright-1.1.362/bin/pyright-langserver --stdio
[jsonrpc] e[17:23:53.221] --> initialize[1] {"jsonrpc":"2.0","id":1,"method":"initialize","params":{"processId":163105,"clientInfo":{"name":"Eglot","version":"1.17"},"rootPath":"/home/matt/tmp/","rootUri":"file:///home/matt/tmp","initializationOptions":{},"capabilities":{"workspace":{"applyEdit":true,"executeCommand":{"dynamicRegistration":false},"workspaceEdit":{"documentChanges":true},"didChangeWatchedFiles":{"dynamicRegistration":true},"symbol":{"dynamicRegistration":false},"configuration":true,"workspaceFolders":true},"textDocument":{"synchronization":{"dynamicRegistration":false,"willSave":true,"willSaveWaitUntil":true,"didSave":true},"completion":{"dynamicRegistration":false,"completionItem":{"snippetSupport":true,"deprecatedSupport":true,"resolveSupport":{"properties":["documentation","details","additionalTextEdits"]},"tagSupport":{"valueSet":[1]}},"contextSupport":true},"hover":{"dynamicRegistration":false,"contentFormat":["markdown","plaintext"]},"signatureHelp":{"dynamicRegistration":false,"signatureInformation":{"parameterInformation":{"labelOffsetSupport":true},"documentationFormat":["markdown","plaintext"],"activeParameterSupport":true}},"references":{"dynamicRegistration":false},"definition":{"dynamicRegistration":false,"linkSupport":true},"declaration":{"dynamicRegistration":false,"linkSupport":true},"implementation":{"dynamicRegistration":false,"linkSupport":true},"typeDefinition":{"dynamicRegistration":false,"linkSupport":true},"documentSymbol":{"dynamicRegistration":false,"hierarchicalDocumentSymbolSupport":true,"symbolKind":{"valueSet":[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26]}},"documentHighlight":{"dynamicRegistration":false},"codeAction":{"dynamicRegistration":false,"resolveSupport":{"properties":["edit","command"]},"dataSupport":true,"codeActionLiteralSupport":{"codeActionKind":{"valueSet":["quickfix","refactor","refactor.extract","refactor.inline","refactor.rewrite","source","source.organizeImports"]}},"isPreferredSupport":true},"formatting":{"dynamicRegistration":false},"rangeFormatting":{"dynamicRegistration":false},"rename":{"dynamicRegistration":false},"inlayHint":{"dynamicRegistration":false},"publishDiagnostics":{"relatedInformation":false,"codeDescriptionSupport":false,"tagSupport":{"valueSet":[1,2]}}},"window":{"showDocument":{"support":true},"workDoneProgress":true},"general":{"positionEncodings":["utf-32","utf-8","utf-16"]},"experimental":{}},"workspaceFolders":[{"uri":"file:///home/matt/tmp","name":"~/tmp/"}]}}
[jsonrpc] e[17:23:53.399] <-- window/logMessage {"jsonrpc":"2.0","method":"window/logMessage","params":{"type":3,"message":"Pyright language server 1.1.362 starting"}}
[jsonrpc] e[17:23:53.399] <-- window/logMessage {"jsonrpc":"2.0","method":"window/logMessage","params":{"type":3,"message":"Server root directory: file:///nix/store/3n04wicy6jpks8qbsqdkjfck1zdwqfpa-pyright-1.1.362/lib/node_modules/pyright/dist"}}
[jsonrpc] e[17:23:53.399] <-- window/logMessage {"jsonrpc":"2.0","method":"window/logMessage","params":{"type":3,"message":"Starting service instance \"~/tmp/\""}}
[jsonrpc] e[17:23:53.400] <-- initialize[1] {"jsonrpc":"2.0","id":1,"result":{"capabilities":{"textDocumentSync":2,"definitionProvider":{"workDoneProgress":true},"declarationProvider":{"workDoneProgress":true},"typeDefinitionProvider":{"workDoneProgress":true},"referencesProvider":{"workDoneProgress":true},"documentSymbolProvider":{"workDoneProgress":true},"workspaceSymbolProvider":{"workDoneProgress":true},"hoverProvider":{"workDoneProgress":true},"documentHighlightProvider":{"workDoneProgress":true},"renameProvider":{"prepareProvider":true,"workDoneProgress":true},"completionProvider":{"triggerCharacters":[".","[","\"","'"],"resolveProvider":true,"workDoneProgress":true,"completionItem":{"labelDetailsSupport":true}},"signatureHelpProvider":{"triggerCharacters":["(",",",")"],"workDoneProgress":true},"codeActionProvider":{"codeActionKinds":["quickfix","source.organizeImports"],"workDoneProgress":true},"executeCommandProvider":{"commands":[],"workDoneProgress":true},"callHierarchyProvider":true,"workspace":{"workspaceFolders":{"supported":true,"changeNotifications":true}}}}}
[jsonrpc] e[17:23:53.400] --> initialized {"jsonrpc":"2.0","method":"initialized","params":{}}
[jsonrpc] e[17:23:53.400] --> textDocument/didOpen {"jsonrpc":"2.0","method":"textDocument/didOpen","params":{"textDocument":{"uri":"file:///home/matt/tmp/foo.py","version":0,"languageId":"python","text":""}}}
[jsonrpc] e[17:23:53.401] --> workspace/didChangeConfiguration {"jsonrpc":"2.0","method":"workspace/didChangeConfiguration","params":{"settings":{}}}
[jsonrpc] e[17:23:53.438] <-- client/registerCapability[0] {"jsonrpc":"2.0","id":0,"method":"client/registerCapability","params":{"registrations":[{"id":"19619ca1-5331-46d6-9e14-77436d5dae53","method":"workspace/didChangeWatchedFiles","registerOptions":{"watchers":[{"globPattern":"**/pyrightconfig.json","kind":7},{"globPattern":"**","kind":7}]}}]}}
[jsonrpc] e[17:23:53.535] --> client/registerCapability[0] {"jsonrpc":"2.0","id":0,"result":null}
[jsonrpc] e[17:23:53.535] <-- workspace/configuration[1] {"jsonrpc":"2.0","id":1,"method":"workspace/configuration","params":{"items":[{"scopeUri":"file:///home/matt/tmp","section":"python"}]}}
[jsonrpc] e[17:23:53.535] --> workspace/configuration[1] {"jsonrpc":"2.0","id":1,"result":[null]}
[jsonrpc] e[17:23:53.537] <-- workspace/configuration[2] {"jsonrpc":"2.0","id":2,"method":"workspace/configuration","params":{"items":[{"scopeUri":"file:///home/matt/tmp","section":"python.analysis"}]}}
[jsonrpc] e[17:23:53.537] --> workspace/configuration[2] {"jsonrpc":"2.0","id":2,"result":[null]}
[jsonrpc] e[17:23:53.538] <-- workspace/configuration[3] {"jsonrpc":"2.0","id":3,"method":"workspace/configuration","params":{"items":[{"scopeUri":"file:///home/matt/tmp","section":"pyright"}]}}
[jsonrpc] e[17:23:53.538] --> workspace/configuration[3] {"jsonrpc":"2.0","id":3,"result":[null]}
[jsonrpc] e[17:23:53.620] <-- window/logMessage {"jsonrpc":"2.0","method":"window/logMessage","params":{"type":3,"message":"No source files found."}}
[jsonrpc] e[17:23:53.641] <-- client/registerCapability[4] {"jsonrpc":"2.0","id":4,"method":"client/registerCapability","params":{"registrations":[{"id":"52ca96a3-8640-46d8-b215-126fd0f42a6f","method":"workspace/didChangeWatchedFiles","registerOptions":{"watchers":[{"globPattern":"**/pyrightconfig.json","kind":7},{"globPattern":"**","kind":7}]}}]}}
[jsonrpc] e[17:23:53.663] --> client/registerCapability[4] {"jsonrpc":"2.0","id":4,"result":null}
[jsonrpc] e[17:23:53.756] <-- textDocument/publishDiagnostics {"jsonrpc":"2.0","method":"textDocument/publishDiagnostics","params":{"uri":"file:///home/matt/tmp/foo.py","version":0,"diagnostics":[]}}
[jsonrpc] e[17:23:53.777] <-- client/unregisterCapability[5] {"jsonrpc":"2.0","id":5,"method":"client/unregisterCapability","params":{"unregisterations":[{"id":"19619ca1-5331-46d6-9e14-77436d5dae53","method":"workspace/didChangeWatchedFiles"}]}}
[jsonrpc] e[17:23:53.777] --> client/unregisterCapability[5] {"jsonrpc":"2.0","id":5,"result":null}
[jsonrpc] e[17:23:53.940] --> textDocument/hover[2] {"jsonrpc":"2.0","id":2,"method":"textDocument/hover","params":{"textDocument":{"uri":"file:///home/matt/tmp/foo.py"},"position":{"line":0,"character":0}}}
[jsonrpc] e[17:23:53.941] --> textDocument/documentHighlight[3] {"jsonrpc":"2.0","id":3,"method":"textDocument/documentHighlight","params":{"textDocument":{"uri":"file:///home/matt/tmp/foo.py"},"position":{"line":0,"character":0}}}
[jsonrpc] e[17:23:53.941] --> textDocument/signatureHelp[4] {"jsonrpc":"2.0","id":4,"method":"textDocument/signatureHelp","params":{"textDocument":{"uri":"file:///home/matt/tmp/foo.py"},"position":{"line":0,"character":0}}}
[jsonrpc] e[17:23:53.948] <-- textDocument/hover[2] {"jsonrpc":"2.0","id":2,"result":null}
[jsonrpc] e[17:23:53.971] <-- textDocument/documentHighlight[3] {"jsonrpc":"2.0","id":3,"result":null}
[jsonrpc] e[17:23:53.971] <-- textDocument/signatureHelp[4] {"jsonrpc":"2.0","id":4,"result":null}
[jsonrpc] e[17:23:54.506] --> workspace/didChangeWatchedFiles {"jsonrpc":"2.0","method":"workspace/didChangeWatchedFiles","params":{"changes":[{"uri":"file:///home/matt/tmp/matt@golem.163105%3A1728424254","type":1}]}}
[jsonrpc] e[17:23:56.235] --> textDocument/didChange {"jsonrpc":"2.0","method":"textDocument/didChange","params":{"textDocument":{"uri":"file:///home/matt/tmp/foo.py","version":12},"contentChanges":[{"range":{"start":{"line":0,"character":0},"end":{"line":0,"character":0}},"rangeLength":0,"text":"i"},{"range":{"start":{"line":0,"character":1},"end":{"line":0,"character":1}},"rangeLength":0,"text":"m"},{"range":{"start":{"line":0,"character":2},"end":{"line":0,"character":2}},"rangeLength":0,"text":"p"},{"range":{"start":{"line":0,"character":3},"end":{"line":0,"character":3}},"rangeLength":0,"text":"o"},{"range":{"start":{"line":0,"character":4},"end":{"line":0,"character":4}},"rangeLength":0,"text":"r"},{"range":{"start":{"line":0,"character":5},"end":{"line":0,"character":5}},"rangeLength":0,"text":"t"},{"range":{"start":{"line":0,"character":6},"end":{"line":0,"character":6}},"rangeLength":0,"text":" "},{"range":{"start":{"line":0,"character":7},"end":{"line":0,"character":7}},"rangeLength":0,"text":"p"},{"range":{"start":{"line":0,"character":8},"end":{"line":0,"character":8}},"rangeLength":0,"text":"a"},{"range":{"start":{"line":0,"character":9},"end":{"line":0,"character":9}},"rangeLength":0,"text":"t"},{"range":{"start":{"line":0,"character":10},"end":{"line":0,"character":10}},"rangeLength":0,"text":"h"},{"range":{"start":{"line":0,"character":11},"end":{"line":0,"character":11}},"rangeLength":0,"text":"l"}]}}
[jsonrpc] e[17:23:56.238] --> textDocument/hover[5] {"jsonrpc":"2.0","id":5,"method":"textDocument/hover","params":{"textDocument":{"uri":"file:///home/matt/tmp/foo.py"},"position":{"line":0,"character":12}}}
[jsonrpc] e[17:23:56.239] --> textDocument/documentHighlight[6] {"jsonrpc":"2.0","id":6,"method":"textDocument/documentHighlight","params":{"textDocument":{"uri":"file:///home/matt/tmp/foo.py"},"position":{"line":0,"character":12}}}
[jsonrpc] e[17:23:56.239] --> textDocument/signatureHelp[7] {"jsonrpc":"2.0","id":7,"method":"textDocument/signatureHelp","params":{"textDocument":{"uri":"file:///home/matt/tmp/foo.py"},"position":{"line":0,"character":12}}}
[jsonrpc] e[17:23:56.244] <-- textDocument/hover[5] {"jsonrpc":"2.0","id":5,"result":null}
[jsonrpc] e[17:23:56.264] <-- textDocument/documentHighlight[6] {"jsonrpc":"2.0","id":6,"result":[{"kind":2,"range":{"start":{"line":0,"character":7},"end":{"line":0,"character":12}}}]}
[jsonrpc] e[17:23:56.264] <-- textDocument/signatureHelp[7] {"jsonrpc":"2.0","id":7,"result":null}
[jsonrpc] e[17:23:56.572] <-- textDocument/publishDiagnostics {"jsonrpc":"2.0","method":"textDocument/publishDiagnostics","params":{"uri":"file:///home/matt/tmp/foo.py","version":12,"diagnostics":[{"range":{"start":{"line":0,"character":7},"end":{"line":0,"character":12}},"message":"Import \"pathl\" could not be resolved","severity":1,"code":"reportMissingImports","source":"Pyright","codeDescription":{"href":"https://github.com/microsoft/pyright/blob/main/docs/configuration.md#reportMissingImports"}},{"range":{"start":{"line":0,"character":7},"end":{"line":0,"character":12}},"message":"\"pathl\" is not accessed","severity":4,"source":"Pyright","tags":[1]}]}}
[jsonrpc] e[17:23:56.660] --> textDocument/completion[8] {"jsonrpc":"2.0","id":8,"method":"textDocument/completion","params":{"textDocument":{"uri":"file:///home/matt/tmp/foo.py"},"position":{"line":0,"character":12},"context":{"triggerKind":1}}}
[jsonrpc] e[17:23:56.711]   <-- textDocument/completion[8] {"jsonrpc":"2.0","id":8,"result":{"items":[{"label":"pathlib","kind":9,"data":{"uri":"file:///home/matt/tmp/foo.py","position":{"line":0,"character":12},"moduleUri":"file:///nix/store/3n04wicy6jpks8qbsqdkjfck1zdwqfpa-pyright-1.1.362/lib/node_modules/pyright/dist/typeshed-fallback/stdlib/pathlib.pyi","symbolLabel":"pathlib"},"sortText":"02.9999.pathlib"}],"isIncomplete":true}}
[jsonrpc] e[17:23:56.712] --> textDocument/didChange {"jsonrpc":"2.0","method":"textDocument/didChange","params":{"textDocument":{"uri":"file:///home/matt/tmp/foo.py","version":13},"contentChanges":[{"range":{"start":{"line":0,"character":12},"end":{"line":0,"character":12}},"rangeLength":0,"text":"ib"}]}}
[jsonrpc] e[17:23:56.712] --> completionItem/resolve[9] {"jsonrpc":"2.0","id":9,"method":"completionItem/resolve","params":{"label":"pathlib","kind":9,"data":{"uri":"file:///home/matt/tmp/foo.py","position":{"line":0,"character":12},"moduleUri":"file:///nix/store/3n04wicy6jpks8qbsqdkjfck1zdwqfpa-pyright-1.1.362/lib/node_modules/pyright/dist/typeshed-fallback/stdlib/pathlib.pyi","symbolLabel":"pathlib"},"sortText":"02.9999.pathlib"}}
[jsonrpc] e[17:23:56.766]   <-- completionItem/resolve[9] {"jsonrpc":"2.0","id":9,"result":{"label":"pathlib","kind":9,"data":{"uri":"file:///home/matt/tmp/foo.py","position":{"line":0,"character":12},"moduleUri":"file:///nix/store/3n04wicy6jpks8qbsqdkjfck1zdwqfpa-pyright-1.1.362/lib/node_modules/pyright/dist/typeshed-fallback/stdlib/pathlib.pyi","symbolLabel":"pathlib"},"sortText":"02.9999.pathlib"}}
[jsonrpc] e[17:23:56.972] --> textDocument/completion[10] {"jsonrpc":"2.0","id":10,"method":"textDocument/completion","params":{"textDocument":{"uri":"file:///home/matt/tmp/foo.py"},"position":{"line":0,"character":14},"context":{"triggerKind":1}}}
[jsonrpc] e[17:23:56.992]   <-- textDocument/publishDiagnostics {"jsonrpc":"2.0","method":"textDocument/publishDiagnostics","params":{"uri":"file:///home/matt/tmp/foo.py","version":13,"diagnostics":[{"range":{"start":{"line":0,"character":7},"end":{"line":0,"character":14}},"message":"\"pathlib\" is not accessed","severity":4,"source":"Pyright","tags":[1]}]}}
[jsonrpc] e[17:23:56.993]   <-- textDocument/completion[10] {"jsonrpc":"2.0","id":10,"result":{"items":[{"label":"pathlib","kind":9,"data":{"uri":"file:///home/matt/tmp/foo.py","position":{"line":0,"character":14},"moduleUri":"file:///nix/store/3n04wicy6jpks8qbsqdkjfck1zdwqfpa-pyright-1.1.362/lib/node_modules/pyright/dist/typeshed-fallback/stdlib/pathlib.pyi","symbolLabel":"pathlib"},"sortText":"01.0000.pathlib"}],"isIncomplete":true}}
[jsonrpc] e[17:23:57.495] --> textDocument/hover[11] {"jsonrpc":"2.0","id":11,"method":"textDocument/hover","params":{"textDocument":{"uri":"file:///home/matt/tmp/foo.py"},"position":{"line":0,"character":14}}}
[jsonrpc] e[17:23:57.495] --> textDocument/documentHighlight[12] {"jsonrpc":"2.0","id":12,"method":"textDocument/documentHighlight","params":{"textDocument":{"uri":"file:///home/matt/tmp/foo.py"},"position":{"line":0,"character":14}}}
[jsonrpc] e[17:23:57.495] --> textDocument/signatureHelp[13] {"jsonrpc":"2.0","id":13,"method":"textDocument/signatureHelp","params":{"textDocument":{"uri":"file:///home/matt/tmp/foo.py"},"position":{"line":0,"character":14}}}
[jsonrpc] e[17:23:57.498] <-- textDocument/hover[11] {"jsonrpc":"2.0","id":11,"result":{"contents":{"kind":"markdown","value":"```python\n(module) pathlib\n```"},"range":{"start":{"line":0,"character":7},"end":{"line":0,"character":14}}}}
[jsonrpc] e[17:23:57.622] <-- textDocument/documentHighlight[12] {"jsonrpc":"2.0","id":12,"result":[{"kind":2,"range":{"start":{"line":0,"character":7},"end":{"line":0,"character":14}}}]}
[jsonrpc] e[17:23:57.622] <-- textDocument/signatureHelp[13] {"jsonrpc":"2.0","id":13,"result":null}
[jsonrpc] e[17:24:15.111] --> workspace/didChangeWatchedFiles {"jsonrpc":"2.0","method":"workspace/didChangeWatchedFiles","params":{"changes":[{"uri":"file:///home/matt/tmp/%23foo.py%23","type":1}]}}
[jsonrpc] e[17:24:15.114] --> workspace/didChangeWatchedFiles {"jsonrpc":"2.0","method":"workspace/didChangeWatchedFiles","params":{"changes":[{"uri":"file:///home/matt/tmp/%23foo.py%23","type":2}]}}
```
</details>

**Comment by joaotavora** [comment]
Thanks.  I think I have everything to try to reproduce.  There's a pending patch to install related to completion, maybe that will solve it.  But the recipe looks good (I'll be trying with `basedpyright` which should be similar)

**Comment by joaotavora** [comment]
I've tried this with `basedpyright` ,then with `pyright` version 1.1.353 and I couldn't reproduce.

You seem to be using a build of upcoming Emacs 30, right? So, in theory, you don't need the `-f package-initialize` to initialize any packages (because python mode is included in Emacs proper).  Can you try without it?

**Comment by joaotavora** [comment]
> and I couldn't reproduce step 7.  

This means I could complete to `import pathlib` and typing `M-TAB` again just leaves it at that, as expected.  I will compare my events log to your later.

**Comment by mcwitt** [comment]
> I've tried this with `basedpyright` ,then with `pyright` version 1.1.353 and I couldn't reproduce step 7.

I'm using pyright 1.1.362.

> You seem to be using a build of upcoming Emacs 30, right? So, in theory, you don't need the `-f package-initialize` to initialize any packages (because python mode is included in Emacs proper). Can you try without it?

I'm able to reproduce with just `emacs -Q` (updated description).

**Comment by joaotavora** [comment]
Can you state exactly what the Git SHA of your Emacs 30 build is?

**Comment by joaotavora** [comment]
@dgutov after the recent changes to `eglot--dumb-tryc` can you make any sense of that backtrace?

**Comment by mcwitt** [comment]
> Can you state exactly what the Git SHA of your Emacs 30 build is?

I'm on `1m1qbdqj1p994wz6flxcswv5c3qqbvjyjfsv0dh65qqq2ph7g4jz`

**Comment by mcwitt** [comment]
Sorry about this; the problem appears to be on my end.

I think this is related to the fact that I had unwittingly been using eglot from ELPA, rather than the built-in package. Doing `M-x list-packages`, eglot shows "external" in the "Status" column. When I remove the external eglot, the same column displays "available" and I'm unable to reproduce the error.

I'm still slightly confused because both the ELPA and built-in packages are version 1.17, but there must be some other difference between the two.

Thanks for the quick response and sorry again for the nosie. Will go ahead and close this unless you think there's anything that still merits investigation.

**Comment by joaotavora** [comment]
No problem.  I'm closing this then.  I find it odd that it still failed for you _without_ `-f package-initialize`  and just `emacs -Q` because I asked you to take it out precisely to rule out that interference from extraneous Eglot packages (installed, presumably, for some older Emacs version)

**Comment by dgutov** [comment]
Could it be that the installed version is actually the newer one? From ELPA, from master. That would still indicate a problem.

**Comment by joaotavora** [comment]
> Could it be that the installed version is actually the newer one? From ELPA, from master. That would still indicate a problem.

Hmmm, indeed it's possible to upgrade... Can you try it out?

**Comment by mcwitt** [comment]
> Hmmm, indeed it's possible to upgrade... Can you try it out?

Confirmed that I can reproduce using the steps above (with or without `-f package-initialize`) if and only if I have eglot 1.17 (commit `b014bca833a17f5b2258e88115f03cffa983d0bd`) in my load path.

**Comment by dgutov** [comment]
@mcwitt Thanks.

@joaotavora Eglot 1.17 predates my changes, and this particular bug is also what you seem to have fixed in the commit `972466dce26`, right after the 1.17 tag.

So maybe the solution is to tag a new version.
